test/rules/switch-default/allow-exhaustive/test.ts.lint

```diff
@@ -0,0 +1,18 @@
+type Tag = 'foo' | 'bar';
+
```

Comment:
should have some tests for enums:

```ts
enum Foo {
    A,
    B,
}
declare let v: Foo;
declare let v2: Foo.A | Foo.B;
```
... and the same with string values enums.

Reply:
Didn't get your point.
The #2104 issue and this PR aren't about enums, they're only about unions. For enums, there is #696 one.

Reply:
Under the hood enums are represented as unions. That means you are fixing both issues here